src/lib/upnp/Discovery.cxx

```diff
@@ -177,12 +177,12 @@ UPnPDeviceDirectory::OnAlive(const UpnpDiscovery *disco) noexcept
 	if (isMSDevice(UpnpDiscovery_get_DeviceType_cstr(disco)) ||
 	    isCDService(UpnpDiscovery_get_ServiceType_cstr(disco))) {
 		try {
-			auto *downloader = new Downloader(*this, *disco);
+			auto downloader = std::make_unique<Downloader>(*this, *disco);
```

Comment:
This change will lead to a crash. Once started, this object must be destroyed only inside the `GetEventLoop()`'s thread (that's why `BlockingCall()` to move the `Destroy()` call into that thread upon error), and only when it's finished. But there are two mistakes here:
- your code unconditionally deletes the object at the end of this scope, leading to use-after-free crash bug
- your code doesn't delete the object in the correct crash, leading to various kinds of race conditions